crates/ruff_cli/src/cache.rs

```diff
@@ -169,7 +203,7 @@ impl Cache {
     ///
     /// This returns `None` if `key` differs from the cached key or if the
     /// cache doesn't contain results for the file.
-    pub(crate) fn get<T: CacheKey>(&self, path: &RelativePath, key: &T) -> Option<&FileCache> {
```

Comment:
I don't think supporting different cache keys for a single cache instance is a good idea. It can lead to key collisions and is prone to mixing different keys (which has the unexpected result that a stored value can not be retrieved). 

E.g. I accidentally passed the hashed `u64` instead of the `FileCacheKey`. Rust was happy because `u64` implements `CacheKey`.

Reply:
Yeah this should be generic on the instance, right? (Looks like it was made non-generic which is also fine with me.)